frontend/src/modules/summary/productList/ProductListContainer.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component, ChangeEvent, ReactNode } from 'react';
+import { Card } from '@material-ui/core';
+
+import ProductList from './ProductList';
+import Product, { getVersionFromString } from 'models/Product';
+import Version from 'models/Version';
+
+interface Props {
+  list: Product[];
+  title: string;
+  handleFavouriteChange(product: Product): void;
+  handlePinChange(product: Product, version: Version, build: string, isPinned: boolean): void;
+}
+
+interface State {
+  searchContent: string;
+  expandedProductsList: string[];
```

Comment:
Philosophical thought... for some of these things that are plain types like strings, would there be benefit in defining actual types instead of using primitive types everywhere? e.g. we could have a type ProductName = string, and then use that everywhere that we deal with a product name via ProductName and ProductName[]. There could be a readability benefit, but i suspect there would be no type checking benefit as I think 2 different subtypes of the same type are (unfortunately) assignable to each other.

Reply:
I would have to look into it but yeah I think you are right. An alternative to that would be to not use primitive types if at all possible. This case could just be the whole product which I think would be fine as long  as we are careful about not having too much stuff in memory.

Side note, I thought I deleted the expanded product list from everywhere as it's no longer needed, I will go and clean it up.